src/Elastic.Apm.Log4Net/Log4NetCorrelation.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Elastic.Apm.Api;
+using log4net;
+
+namespace Elastic.Apm.Log4Net
+{
+	public class Log4NetCorrelation : ITransactionObserver
+	{
+		public void TransactionStarted(ITransaction transaction)
+		{
+			if (!Agent.IsConfigured) return;
+
+			var currentTransaction = Agent.Tracer.CurrentTransaction;
+
+			if (currentTransaction == null)
+			{
+				LogicalThreadContext.Properties.Remove("Transaction.Id");
+				LogicalThreadContext.Properties.Remove("Trace.Id");
+				return;
+			}
+
+			LogicalThreadContext.Properties["Transaction.Id"] = currentTransaction.Id;
```

Comment:
Maybe it's worth using more globally unique names (for example `"Elastic.Apm.Transaction.Id"`) to avoid clashing with other libraries and/or the application?

Reply:
This follows the elastic common schema.